### ISSUE TITLE
xval example code is incorrect

### ISSUE THREAD

**Issue by johann-petrak** [deepset-ai/FARM#812]
The current example program `./examples/doc_classification_crossvalidation.py` is calculating the overal XVAL estimates incorrectly. For each metric, it should really calculate the mean and stdev over all folds instead of calculating the metric over all fold data.  In most cases, the difference is probably tiny, but the way this is calculated now is definitely wrong.

**Comment by johann-petrak** [comment]
Metrics logged for the current implementation (single run, seed 42): 

```
TEST F1 MICRO:   0.9147791619479049
TEST F1 MACRO:   0.9063262299015191
TEST F1 OFFENSE: 0.8781869688385269
TEST F1 OTHER:   0.9344654909645113
TEST MCC:        0.8134335820565264
```

**Comment by johann-petrak** [comment]
New implementation:

```
XVAL Accuracy:   mean 0.8021317093826182 stdev 0.010111693788346669
XVAL F1 MICRO:   mean 0.8021317093826182 stdev 0.010111693788346709
XVAL F1 MACRO:   mean 0.7816765438006879 stdev 0.008313385095785861
XVAL F1 OFFENSE: mean 0.7151650412974621 stdev 0.010665611398946019
XVAL F1 OTHER:   mean 0.8481880463039135 stdev 0.010688811556707749
XVAL MCC:        mean 0.5656135088223573 stdev 0.017285658033477026
```

**Comment by Timoeller** [comment]
fixed by #825
